en/setup/citationkeypatterns.md

```diff
@@ -18,57 +18,79 @@ The default key pattern is **`[auth][year]`**, and this could produce keys like
+* **`[edtrshort]`**: The last name if one editor is given; the first character of up to three editors' last names if more than one editor is given. A plus character is added, if there are more than three editors
+* **`[edtrForeIni]`**: The forename initial of the first editor
+* **`[editorLastForeIni]`**: The forename initial of the last editor
+
+#### Title-related field markers
+
+* **`[shorttitle]`**: The first 3 words of the title, ignoring any function words \(see below\). For example, `An awesome paper on JabRef` becomes `AwesomePaperJabref`
+* **`[shorttitleINI]`**: The first 3 words of the title, abbreviated.
+* **`[veryshorttitle]`**: The first word of the title, ignoring any function words \(see below\). For example, `An awesome paper on JabRef` becomes `Awesome`
+* **`[camel]`**: Capitalize and concatenate all the words of the title. For example, `An awesome paper on JabRef` becomes `AnAwesomePaperOnJabref`
+* **`[title]`**: Capitalize all the significant words of the title, and concatenate them. For example, `An awesome paper on JabRef` becomes `AnAwesomePaperonJabref`
+* **`[fulltitle]`**: The title with unchanged capitalization.
 
 JabRef considers the following words to be [function words](https://en.wikipedia.org/wiki/Function_word): "a", "an", "the", "above", "about", "across", "against", "along", "among", "around", "at", "before", "behind", "below", "beneath", "beside", "between", "beyond", "by", "down", "during", "except", "for", "from", "in", "inside", "into", "like", "near", "of", "off", "on", "onto", "since", "to", "toward", "through", "under", "until", "up", "upon", "with", "within", "without", "and", "but", "for", "nor", "or", "so", "yet".
 
-#### Other key patterns
+#### Other field markers
 
+* **`[entrytype]`**: The type of the entry, e.g., `Article`, `InProceedings`, etc
 * **`[firstpage]`**: The number of the first page of the publication \(Caution: this will return the lowest number found in the pages field, i.e. for `7,41,73--97` it will return `7`.\)
-* **`[pageprefix]`**: The non-digit prefix of pages \(like "L" for L7\) or "" if no non-digit prefix exists \(like "" for `7,41,73--97`\) .
-* **`[keywordN]`**: Keyword number N from the “keywords” field, assuming keywords are separated by commas or semicolons.
+* **`[pageprefix]`**: The non-digit prefix of pages \(like `L` for `L7`\) or "" if no non-digit prefix exists \(like "" for `7,41,73--97`\)
+* **`[keywordN]`**: Keyword number `N` from the “keywords” field, assuming keywords are separated by commas or semicolons
+* **`[keywordsN]`**: Up to `N` keywords from the "keywords" field
 * **`[lastpage]`**: The number of the last page of the publication \(See the remark on `firstpage`\)
 * **`[shortyear]`**: The last 2 digits of the publication year
 
+#### Bibentry fields
+
+In addition to the special field markers, most BibTeX, biblatex, and JabRef field names can be used directly. If you regularly use a field name not on this list, you are encouraged to add it.
```

Comment:
Shouldn't this come first, say before the section "Default citation key pattern". So first we describe the normal usage of fields, and then more specialized treatment of author and title etc? 

(Also I would move the "Default citation key pattern" section down to the "how to configure" section.

Reply:
I think there are a few different issues here to unpack,

1. Should there be a different help document for the bracketed patterns? In which case, sorting/splitting "field markers" ("field expressions" is more descriptive?) will allow for a quick lookup without having the description of what it is and where/how they are used. It would also make it easier to avoid duplications for issue #265 .
2. I'd expect the bibentry fields to be the least used ones, as their use are likelier to differ from what is being shown in the JabRef GUI (and some bibentry fields are overridden by "special fields (e.g., `title` and `entrytype`)).
3. I agree that the "Default citation pattern" belongs close to the "How to configure section". I'll move it, and update the "How to configure" section.

Reply:
I agree! What about combining the two things:
1. Describe the general structure of the pattern in more detail at the beginning. It's talking about `[volume]` a bit, but from that description it doesn't really come across that you can use essential every field there.
2. Then describe the most-used special fields like author and title.
3. Then describe the remaining special fields like entrytype or groups. 

In general, the help should be structured so that new users get a pretty good idea about the feature from the first paragraph/section, and then more information follows for the experts.

Reply:
How long is too long for a help page? I'd consider this an option,

1. Describe the bracketed pattern, I quite like [the current section](https://docs.jabref.org/setup/citationkeypatterns#key-patterns) but perhaps add a few more interesting examples
2. Describe field markers by field type (Author, Title, Others...)
3. Describe how this applies to citation keys (including what is the default and how to change it)
4. Describe how this applies to file patterns (including how to change them)
5. Describe how they are used in file searches (I don't know this part yet so I am not sure of what should be included)

But I'd also consider to have 1-2 on its own page (help page about bracketed patterns), and then 3, 4, and 5 on their own page, with more examples and a link to 2.

Reply:
Sounds good! I would keep 1-3 under citation keys, and add a remark for 4 in the help page that talks about the file management (along the lines of "the pattern is the same as for the citation key, e.g. `[bibtex] - [title] gives .... For more details, see xyz").

Reply:
I'll get to it! I'll search the discourse forum to see if I can find some "real" examples (including the "-" issue)